### PR TITLE
defer classes to talk to gateway

### PR DESCRIPTION
**Pull request by sbarker2**

### CODE REVIEW

**Review by felipe-gdr — APPROVED** [review]

File: lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt [hunk 1]
```diff
@@ -1,7 +1,16 @@
 package graphql.nadel
 
+import graphql.incremental.DelayedIncrementalPartialResult
+import org.reactivestreams.Publisher
+
 open class ServiceExecutionResult @JvmOverloads constructor(
```
Comment (felipe-gdr) [inline lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt:6]: can we make `ServiceExecutionResult` `sealed`, and _not_ make `NadelncrementalServiceExecutionResult` `open`?

Comment (sbarker2) [reply lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt:6]: oh yeah, forgot to change that back

**Review by bbakerman — DISMISSED** [review]

File: lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt (hunk 1 above)
Comment (sbarker2) [reply lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt]: @felipe-gdr Hmm, I'm not sure if this is what we want actually. Sealed types cannot be instantiated, but it needs to be instantiated here in IntrospectionService.kt

https://github.com/atlassian-labs/nadel/blob/9897a5db6024cd179c8b1840ee9043448618cb7c/lib/src/main/java/graphql/nadel/engine/blueprint/IntrospectionService.kt#L48C1-L51C18

Comment (felipe-gdr) [reply lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt]: Ah right
[A sealed class is [abstract](https://kotlinlang.org/docs/classes.html#abstract-classes) by itself, it cannot be instantiated directly](https://kotlinlang.org/docs/sealed-classes.html#location-of-direct-subclasses)

We could create a `ServiceExecutionResultImpl` class that extends it and instantiate that instead.

Comment (sbarker2) [reply lib/src/main/java/graphql/nadel/ServiceExecutionResult.kt]: Changed, thanks

**Review by felipe-gdr — APPROVED** [review]

**Pull request merged**